File: platform/polyglot/saucelabs_api/py/saucelabs_api/modules/platform.py
"""
Platform Module — Sauce Labs API Client

Check service status and supported configurations.

Endpoints:
    GET /rest/v1/info/status                        - Service status
    GET /rest/v1/info/platforms/{automation_api}     - Supported platforms
"""

from __future__ import annotations

from typing import Any

from ..errors import SaucelabsValidationError
from ..logger import create_logger
from ..types import AUTOMATION_API_VALUES

log = create_logger("saucelabs_api", "platform")


class PlatformModule:
    """Query Sauce Labs platform status and capabilities."""

    def __init__(self, client: Any) -> None:
        self._client = client
        self._logger = log

    async def get_status(self) -> dict[str, Any]:
        """Get current Sauce Labs service status (public, no auth required)."""
        self._logger.debug("checking service status")
        return await self._client.get("/rest/v1/info/status")

    async def get_platforms(self, automation_api: str = "all") -> list[dict[str, Any]]:
        """Get supported platforms filtered by automation backend.

        Parameters
        ----------
        automation_api:
            Filter: 'all', 'appium', or 'webdriver'.
        """
        if automation_api not in AUTOMATION_API_VALUES:
            raise SaucelabsValidationError(
                f"automation_api must be one of: {', '.join(AUTOMATION_API_VALUES)} — got '{automation_api}'"
            )

        self._logger.debug("getting platforms", {"automation_api": automation_api})
        return await self._client.get(f"/rest/v1/info/platforms/{automation_api}")

File: platform/polyglot/saucelabs_api/mjs/src/modules/platform.mjs
/**
 * Platform Module — Sauce Labs API Client
 *
 * Check service status and supported configurations.
 *
 * Endpoints:
 *   GET /rest/v1/info/status                        - Service status
 *   GET /rest/v1/info/platforms/{automation_api}     - Supported platforms
 */

import { create } from '../logger.mjs';
import { SaucelabsValidationError } from '../errors.mjs';
import { AUTOMATION_API_VALUES } from '../types.mjs';

const log = create('saucelabs-api', import.meta.url);

export class PlatformModule {
  /**
   * @param {import('../client.mjs').SaucelabsClient} client
   */
  constructor(client) {
    this._client = client;
    this._logger = log;
  }

  /**
   * Get current Sauce Labs service status.
   * This is a public endpoint — no authentication required.
   *
   * @returns {Promise<object>} Service status object
   */
  async getStatus() {
    this._logger.debug('checking service status');
    return this._client.get('/rest/v1/info/status');
  }

  /**
   * Get supported platforms filtered by automation backend.
   * This is a public endpoint — no authentication required.
   *
   * @param {string} automationApi - Filter: 'all', 'appium', or 'webdriver'
   * @returns {Promise<Array>} List of platform configurations
   */
  async getPlatforms(automationApi = 'all') {
    if (!AUTOMATION_API_VALUES.includes(automationApi)) {
      throw new SaucelabsValidationError(
        `automationApi must be one of: ${AUTOMATION_API_VALUES.join(', ')} — got "${automationApi}"`,
      );
    }

    this._logger.debug('getting platforms', { automationApi });
    return this._client.get(`/rest/v1/info/platforms/${automationApi}`);
  }
}
